pkg/controller/elasticsearch/client/client.go

```diff
@@ -9,19 +9,28 @@ import (
 	"crypto/x509"
 	"encoding/json"
 	"fmt"
+	"math"
 	"net/http"
 	"time"
 
+	"k8s.io/apimachinery/pkg/runtime"
+
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/annotation"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
 	"github.com/elastic/cloud-on-k8s/pkg/utils/net"
 )
 
 const (
-	// DefaultVotingConfigExclusionsTimeout is the default timeout for setting voting exclusions.
-	DefaultVotingConfigExclusionsTimeout = "30s"
```

Comment:
Somewhat unrelated to that PR: I could not remember what that timeout was for so I double-checked the code.
It is used as a query param in the voting config exclusions request: https://www.elastic.co/guide/en/elasticsearch/reference/master/voting-config-exclusions.html#voting-config-exclusions-api-query-params.
If the timeout is reached, the request fails and returns an error.

Other API calls, such as `GET _cluster/settings` (https://www.elastic.co/guide/en/elasticsearch/reference/master/cluster-get-settings.html) also have this timeout query param, but we don't use it (we should!).

For the sake of simplicity (one less thing to configure), I feel like we could get rid of that `DefaultVotingConfigExclusionsTimeout ` and use the `defaultClientTimeout` instead (which moves the timeout to 3min instead of the default 30sec). Since we'd also have a 3min timeout to the http call, we may reach it before Elasticsearch reaches its own 3min timeout to set the voting config exclusions. Which is not a big deal, I think: this request is idempotent and we do it at every reconciliation anyway.

To summarize:
- keep a single Elasticsearch http timeout for all api calls
- ensure we set the `?timeout=` query param to that timeout for URLs that support it

@elastic/cloud-k8s any thoughts?

Reply:
I am in favour of your proposal. I have to admit I did not question the need for this timeout in the first place.

Reply:
Currently we don't use the `timeout` override in the API call at all (we just default to the Elasticsearch default) so I have opted to remove that parameter completely from the function for now. If we find a reason to use an explicit timeout at  a later date, we can easily add that back in.